doc/langref.html.in

```diff
@@ -5088,6 +5088,46 @@ fn deferUnwindExample() void {
 
 test "defer unwinding" {
     deferUnwindExample();
+}
+      {#code_end#}
+      {#code_begin|test|test_mutable_vars_defer#}
+const std = @import("std");
+
+const expect = std.testing.expect;
+
+// It's important to understand the semantics of `defer` when using mutable
+// variables. The `defer` statement captures the value of the variable at the
+// time the `defer` statement is executed, which is at the end of the current
+// scope. This means that if the variable is mutated after the `defer` statement
+// is defined but before the end of the scope, the `defer` statement will
+// execute with the mutated value.
```

Comment:
In my opinion, it's confusing to speak of `defer` as capturing values. Here's an example of how it could be worded otherwise:
```suggestion
// It's important to understand the semantics of `defer` when using mutable
// variables. Because the `defer` statement executes at the end of the scope, 
// the variables used by the `defer` statement have the values they have at 
// the end of the scope, not when the `defer` statement is defined.
```